osf/external/oopspam/tasks.py

```diff
@@ -0,0 +1,37 @@
+from framework import sentry
+from framework.celery_tasks import app as celery_app
+
+from osf.external.oopspam.client import OOPSpamClient
+from osf.external.oopspam.exceptions import OOPSpamClientError
+
+
+@celery_app.task()
+def check_resource_with_oopspam(guid, content, author, author_email, request_kwargs):
+    from osf.models import Guid
+    resource = Guid.load(guid).referent
+    client = OOPSpamClient()
+
+    try:
+        is_spam, oopspam_details = client.check_content(
+            user_ip=request_kwargs['remote_addr'],
+            content=content
+        )
+    except OOPSpamClientError:
+        sentry.log_exception()
+        return
+
+    resource.spam_data['headers'] = request_kwargs
+    resource.spam_data['content'] = content
+    resource.spam_data['author'] = author
+    resource.spam_data['author_email'] = author_email
+    resource.spam_data['oopspam_data'] = oopspam_details
+    if is_spam:
+        resource.flag_spam()
+        if not resource.spam_data.get('who_flagged'):
+            resource.spam_data['who_flagged'] = 'oopspam'
+        elif resource.spam_data['who_flagged'] == 'akismet':
+            resource.spam_data['who_flagged'] = 'both'
```

Comment:
DRY. Why not have a single `check_spam_task` which does something like
``` python
any_is_spam = False
for client in SPAM_CLIENTS:
    is_spam, details = client.check_content(*args)
    if is_spam:
        any_is_spam = True
        resource.spam_data[client.name]['details'] = details
        #...

if any_is_spam:
    resource.flag_spam()
    who_flagged = list(resource.spam_data.keys())
    resource.spam_data['who_flagged'] = who_flagged[0] if len(who_flagged) == 1 else 'both'  
    #could alternatively just have list here in case we add other clients in the future

resource.save()
# no need for return because this is a task
```
Otherwise you've got multiple tasks competing for the same fields on the same resource (lock-prone), duplicated logic, twice as many tasks as necessary to process.

This may require tweaking the `check_content` signatures and the `__init__` for the clients, though `AkismetClient.__init__` just uses `settings.` args, which ought be used as defaults in the fn.